src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/response/Commit.java

```diff
@@ -0,0 +1,13 @@
+package com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.response;
+
+public class Commit {
+    private String id;
```

Comment:
Make this `final` and initialize it in the constructor. If it's being set by JSON deserialisation, then use `@JsonCreator` and `@JsonProperty`, e.g.
```java
    @JsonCreator
    public Commit(@JsonProperty("id") String id) {
        super();
        this.id = id;
    }
````